src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -246,6 +246,11 @@ void java_bytecode_convert_classt::convert(
 /// java::array[X] where X is byte, float, int, char...
 void java_bytecode_convert_classt::add_array_types()
 {
+  // this method does not need to be called more than once
+  static bool already_executed=false;
+  if(already_executed)
+    return;
+
```

Comment:
I fully understand the need for a fix, and fixing this is very good. But it seems it should be properly architected rather than doing this sort of band-aid fix. A proper architecture will ensure that the method can only be called once, which generally is done via constructors. In this case my suspicion is such that the method is simply in the wrong class: if it shouldn't be call per class, why could it possibly be part of `convert_class`?